source/agora/api/handler/PreImageReceivedHandler.d

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+
+    Definitions of the PreImageReceivedHandler
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.api.handler.PreImageReceivedHandler;
+
+import agora.consensus.data.PreImageInfo;
+
+import vibe.web.rest;
+import vibe.http.common;
+
+public interface PreImageReceivedHandler
+{
+// The REST generator requires @safe methods
+@safe:
```

Comment:
Is this really necessary?

Reply:
> Is this really necessary?

Don't you see the comment? It says `REST generator requires @safe`